backend/routes/nft.go

```diff
@@ -195,3 +206,143 @@ func mintNFTDevnet(w http.ResponseWriter, r *http.Request) {
+	}
+	defer rows.Close()
+
+	if !rows.Next() {
+		// No rows returned, user hasn't liked the NFT yet
+		_, err = core.ArtPeaceBackend.Databases.Postgres.Exec(context.Background(), "INSERT INTO NFTLikes (nftKey, liker) VALUES ($1, $2)", nftlikeReq.NFTKey, nftlikeReq.UserAddress)
+		if err != nil {
+			http.Error(w, errors.Wrap(err, "error inserting into database").Error(), http.StatusInternalServerError)
+			return
+		}
+		fmt.Println("NFT Liked By User")
+		w.WriteHeader(http.StatusOK)
+		w.Write([]byte("NFT Liked By User"))
+		return
+	}
+
+	// User has already liked the NFT
+	fmt.Println("NFT Already Liked By User")
+	w.WriteHeader(http.StatusOK)
+	w.Write([]byte("NFT Already Liked By User"))
+}
+
+func UnLikeNFT(w http.ResponseWriter, r *http.Request) {
+
+	w.Header().Set("Access-Control-Allow-Origin", "*")
+	w.Header().Set("Content-Type", "application/json")
+
+	if r.Method != http.MethodPost {
+		http.Error(w, "Method not allowed", http.StatusMethodNotAllowed)
+		return
+	}
+
+	var nftlikeReq NFTLikesRequest
+	err := json.NewDecoder(r.Body).Decode(&nftlikeReq)
+	if err != nil {
+		http.Error(w, errors.Wrap(err, "error decoding JSON").Error(), http.StatusBadRequest)
+		return
+	}
+
+	log.Printf("Received request: %+v", nftlikeReq)
+
+	// check if the user has like the nft
+	rows, err := core.ArtPeaceBackend.Databases.Postgres.Query(context.Background(), "SELECT * FROM NFTLikes WHERE nftKey = $1 AND liker = $2", nftlikeReq.NFTKey, nftlikeReq.UserAddress)
+
+	w.Header().Set("Access-Control-Allow-Origin", "*")
+	w.Header().Set("Content-Type", "application/json")
+	w.WriteHeader(http.StatusOK)
+
+	defer rows.Close()
+
+	if !rows.Next() { // No rows returned, user hasn't liked the NFT yet
+
+		w.WriteHeader(http.StatusOK)
+		w.Write([]byte("Cannot Unlike an NFT You did not Like"))
+		fmt.Println("Cannot Unlike an NFT You did not Like")
+		return
+
+	}
+
+	// delete the like here
+	_, err = core.ArtPeaceBackend.Databases.Postgres.Query(context.Background(), "DELETE FROM nftlikes WHERE nftKey = $1 AND liker = $2", nftlikeReq.NFTKey, nftlikeReq.UserAddress)
+
+	if err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		w.Write([]byte(err.Error()))
+		return
+	}
+
+	w.WriteHeader(http.StatusOK)
+	w.Write([]byte("NFT Unlike By User"))
+	fmt.Println("NFT Unlike By User")
+	return
+
+}
+
+func getNftlikeCounts(w http.ResponseWriter, r *http.Request) {
+
+	nftkey := r.URL.Query().Get("nft_key")
+
+	// check nftkey is not empty
+
+	if nftkey == "" {
+		http.Error(w, `{"error": "Missing nft key parameter"}`,
+			http.StatusBadRequest)
+		return
+	}
+
+	w.Header().Set("Access-Control-Allow-Origin", "*")
+	w.Header().Set("Content-Type", "application/json")
+	w.WriteHeader(http.StatusOK)
+
+	log.Printf("nftKey: %d", nftkey)
+
+	var count int
+
+	err := core.ArtPeaceBackend.Databases.Postgres.QueryRow(context.Background(), "SELECT COUNT(*) FROM nftlikes WHERE nftKey = $1", nftkey).Scan(&count)
+
+	if err != nil {
+
+		if err == sql.ErrNoRows {
```

Comment:
Please change all `sql` library usage to the `pgx` library, like done here : https://github.com/keep-starknet-strange/art-peace/blob/f32dba235655fa64d583d80e720ab2d9282bdecb/backend/routes/user.go#L31

https://github.com/keep-starknet-strange/art-peace/blob/f32dba235655fa64d583d80e720ab2d9282bdecb/backend/routes/user.go#L8